spark2/src/test/java/org/apache/iceberg/spark/source/TestCatalog.java

```diff
@@ -95,7 +95,7 @@ public void initialize(String name, Map<String, String> properties) {
     String uri = properties.get(CatalogProperties.URI);
     warehouse = properties.get("warehouse");
     Preconditions.checkArgument(uri != null, "A uri parameter must be set");
-    Preconditions.checkArgument(uri.contains("thrift"), "A ur parameter must be valid");
+    Preconditions.checkArgument(uri.contains("thrift"), "A uri parameter must be valid");
```

Comment:
While we're here, should we consider changing this to something more explanatory?

Perhaps `Catalog uri parameter must be of type thrift` would be better?

I don't have a strong opinion either way, since it's just a test. But given that I'm here already, figured I'd bring it up. I strongly prefer informative error messages over more vague error messages.

Reply:
I agree @kbendick a more informative message would be better. @rdblue always asks for 'cannot do X, reason. Optionally how to fix'. I think that's a great standard for the project, even in tests.

Reply:
How about `Cannot initialize TestCatalog, the catalog's metastore connection uri must use thrift as the scheme.`?

The optional fix, in theory, would be something along the lines of `Please update the configuration on the SparkContext for 'spark.sql.catalog.${catalog_name}.uri' to the form of 'thrift://host:port'`, though in this case, I think that the optional how to fix is likely unnecessary (as it's just a mock, though that is how one would resolve that issue).

What do you think? I'm tempted to update all of these Preconditions while I'm here, as I feel like people often look at the tests to learn how to do things (even if they could realistically get that information from the docs). 🙂

Reply:
> How about `Cannot initialize TestCatalog, the catalog's metastore connection uri must use thrift as the scheme.`?

agreed, the fix is probably redundant for the tests

>  I'm tempted to update all of these Preconditions while I'm here, 
Love the idea, thanks for taking that on. makes a massive difference to the community imho. I always start w/ reading the tests :-)

Reply:
Always happy to help make the project more friendly to newcomers 👍.

I'll update the Preconditions in the morning (it's admittedly really later where I am).

I've got some other, much larger and long overdue community help stuff lined up too (a docker-compose environment) that I am hoping to push next week. I'd love to get it working first for the basics (spark with a consistent metastore - currently barebones Hive) and then add Nessie to it as well 🙂. I'm currently in the process of removing the internal bits and then deciding on what is too much, but will probably just start small and iterate. I'll tag you there and might reach out to ask for your opinion as well if you don't mind.